pkg/controller/servicebindingrequest/planner.go

```diff
@@ -0,0 +1,166 @@
+package servicebindingrequest
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	apiextv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ustrv1 "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+// Planner plans resources needed to bind a given backend service, using OperatorLifecycleManager
+// standards and CustomResourceDefinitionDescription data to understand which attributes are needed.
+type Planner struct {
+	ctx    context.Context                 // request context
+	client client.Client                   // Kubernetes API client
+	ns     string                          // namespace name
+	sbr    *v1alpha1.ServiceBindingRequest // instantiated service binding request
+	logger logr.Logger                     // logger instance
+}
+
+// Plan outcome, after executing planner.
+type Plan struct {
+	Ns             string                      // namespace name
+	Name           string                      // plan name, same than ServiceBindingRequest
+	CRDDescription *olmv1alpha1.CRDDescription // custom resource definition description
+	CRD            *ustrv1.Unstructured        // custom resource definition
+}
+
+const (
+	connectsToLabel = "connects-to"
+)
+
+// extractConnectTo inspect ServiceBindingRequest to extract connects-to value.
+func (p *Planner) extractConnectsTo() string {
```

Comment:
@otaviof 

```
matchLabels:
      connects-to: postgres
      environment: production
```

The hard requirement for having a specific label should not be done. That would make it a strong constraint. Any label should be supported here for selecting objects. We can take care of this in a separate PR though.

Reply:
Indeed, I think we can address this in a stand-along PR, which could also include E2E coverage started on this PR.